### PR TITLE
Add component metadata YAML for configuration.redis.

### PR DESCRIPTION
**Pull request by tmacam**
# Description

Add component metadata YAML for configuration.redis.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: configuration/redis/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+status: alpha
+title: "Redis"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-configuration-stores/redis-configuration-store/
+capabilities: []
+metadata:
+  - name: redisHost
+    required: true
+    description: Connection-string for the redis host
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:15]:
```suggestion
    description: Connection string for the Redis host
```

File: configuration/redis/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    example: "redis-master.default.svc.cluster.local:6379"
+    type: string
+  - name: redisPassword
+    required: false
+    description: Password for Redis host. No Default. Can be secretKeyRef to use a secret reference
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:20]:
```suggestion
    description: Password for the Redis host
```

File: configuration/redis/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    example:  "KeFg23!"
+    type: string
+  - name: enableTLS
+    required: false
+    description: | 
+      If the Redis instance supports TLS with public certificates, can be
+      configured to be enabled or disabled. Defaults to false.
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:27]:
```suggestion
      If enabled, connects to Redis using TLS with public certificates.
```

File: configuration/redis/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    required: false
+    description: |
+      Maximum number of retries before giving up. Defaults to "3".
+    default: "3"
+    example: "5"
+    type: number
+  - name: maxRetryBackoff
+    required: false
+    description: |
+      Maximum backoff between each retry in milliseconds. Defaults to 2 seconds; "-1" disables backoff.
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:41]: We really need to change these durations, nanoseconds are ridiculous...

```suggestion
      Maximum backoff between each retry in nanoseconds. Defaults to 2 seconds; "-1" disables backoff.
```

File: configuration/redis/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    type: number
+  - name: failover
+    required: false
+    description: |
+      Property to enabled failover configuration. Needs sentinelMasterName to
+      be set. The redisHost should be the sentinel host address. See Redis
+      Sentinel Documentation. Defaults to "false".
+    default: "false"
+    example: "true"
+    type: bool
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:53]:
```suggestion
  - name: failover
    required: false
    description: |
      Enables failover configuration. It requires "sentinelMasterName" to
      be set, and "redisHost" to be the sentinel host address.
    default: "false"
    example: "true"
    type: bool
    url:
      title: "Redis Sentinel documentation"
      url: "https://redis.io/docs/manual/sentinel/"
```

File: configuration/redis/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  - name: sentinelMasterName
+    required: false
+    description: |
+      The sentinel master name. Requires "failover" to be set. See Redis
+      Sentinel Documentation.
+    example:  "127.0.0.1:6379"
+    type: string
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:60]:
```suggestion
  - name: sentinelMasterName
    required: false
    description: |
      The Redis sentinel master name. Required when "failover" is enabled.
      Sentinel Documentation.
    example:  "127.0.0.1:6379"
    type: string
```

File: configuration/redis/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  - name: redisDB
+    required: false
+    description: |
+      Database selected after connecting to redis. If "redisType" is "cluster"
+      this option is ignored. Defaults to "0".
```
Comment (ItalyPaleAle) [inline configuration/redis/metadata.yaml:65]:
```suggestion
      Database selected after connecting to redis. If "redisType" is "cluster"
      this option is ignored.
```

File: configuration/redis/metadata.yaml (hunk 4 above)
Comment (tmacam) [reply configuration/redis/metadata.yaml:41]: I know :) And it is not even uniform how we handle those among the many Redis components we have.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**